pkg/combustion/rpm_test.go

```diff
@@ -11,158 +13,193 @@ import (
+func TestWriteRPMScriptWithRPMRepo(t *testing.T) {
+	ctx, teardown := setupContext(t)
+	defer teardown()
 
-	file1, err := os.Create(filepath.Join(rpmSourceDir, "rpm1.rpm"))
+	repoName := "foo"
+	pkgList := []string{"pkg1", "pkg2", "pkg3"}
+	script, err := writeRPMScript(ctx, repoName, pkgList)
 	require.NoError(t, err)
+	assert.Equal(t, modifyRPMScriptName, script)
 
-	file2, err := os.Create(filepath.Join(rpmSourceDir, "rpm2.rpm"))
+	expectedFilename := filepath.Join(ctx.CombustionDir, modifyRPMScriptName)
+	foundBytes, err := os.ReadFile(expectedFilename)
 	require.NoError(t, err)
 
-	return ctx, rpmSourceDir, func() {
-		assert.NoError(t, file1.Close())
-		assert.NoError(t, file2.Close())
+	stats, err := os.Stat(expectedFilename)
+	require.NoError(t, err)
+	assert.Equal(t, fileio.ExecutablePerms, stats.Mode())
 
-		teardownCtx()
-	}
+	foundContents := string(foundBytes)
+	zypperAR := fmt.Sprintf("zypper ar file://%s %s", filepath.Join(combustionBasePath, repoName), repoName)
+	zypperInstall := fmt.Sprintf("zypper --no-gpg-checks install -r %s -y --force-resolution --auto-agree-with-licenses %s", repoName, strings.Join(pkgList, " "))
+	zypperRR := fmt.Sprintf("zypper rr %s", repoName)
+	assert.Contains(t, foundContents, zypperAR)
+	assert.Contains(t, foundContents, zypperInstall)
+	assert.Contains(t, foundContents, zypperRR)
 }
 
-func TestGetRPMFileNames(t *testing.T) {
-	// Setup
-	_, rpmSourceDir, teardown := setupRPMSourceDir(t)
+func TestWriteRPMScriptStandaloneRPM(t *testing.T) {
+	ctx, teardown := setupContext(t)
 	defer teardown()
 
-	// Test
-	rpmFileNames, err := getRPMFileNames(rpmSourceDir)
+	repoName := ""
+	pkgList := []string{"pkg1", "pkg2", "pkg3"}
+	script, err := writeRPMScript(ctx, repoName, pkgList)
+	require.NoError(t, err)
+	assert.Equal(t, modifyRPMScriptName, script)
 
-	// Verify
+	expectedFilename := filepath.Join(ctx.CombustionDir, modifyRPMScriptName)
+	foundBytes, err := os.ReadFile(expectedFilename)
 	require.NoError(t, err)
 
-	assert.Contains(t, rpmFileNames, "rpm1.rpm")
-	assert.Contains(t, rpmFileNames, "rpm2.rpm")
+	stats, err := os.Stat(expectedFilename)
+	require.NoError(t, err)
+	assert.Equal(t, fileio.ExecutablePerms, stats.Mode())
+
+	foundContents := string(foundBytes)
+	zypperAR := "zypper ar file:/"
+	zypperInstall := fmt.Sprintf("zypper --no-gpg-checks install -y --force-resolution --auto-agree-with-licenses %s", strings.Join(pkgList, " "))
+	zypperRR := "zypper rr"
+	assert.Contains(t, foundContents, zypperInstall)
+	assert.NotContains(t, foundContents, zypperAR)
+	assert.NotContains(t, foundContents, zypperRR)
 }
 
-func TestCopyRPMs(t *testing.T) {
-	// Setup
-	ctx, rpmSourceDir, teardown := setupRPMSourceDir(t)
+func TestWriteRPMScriptEmptyPKGList(t *testing.T) {
+	_, err := writeRPMScript(nil, "", []string{})
+	require.Error(t, err)
+	require.ErrorContains(t, err, "package list cannot be empty")
+}
+
+func TestSkipRPMConfigurePositive(t *testing.T) {
```

Comment:
Tests shouldn't require on inline comments as names for the scenarios. Can we either introduce table testing for skipping the component and validating the resolution or split up the assertions in multiple smaller functions where each one describes what it is testing?